src/client.js

```diff
@@ -109,6 +117,73 @@ export class Saturn {
     return { res, controller, log }
   }
 
+  /**
+   *
+   * @param {string} cidPath
+   * @param {object} [opts={}]
+   * @param {('car'|'raw')} [opts.format]
+   * @param {string} [opts.url]
+   * @param {number} [opts.connectTimeout=5000]
+   * @param {number} [opts.downloadTimeout=0]
+   * @returns {Promise<AsyncIterable<Uint8Array>>}
+   */
+  async * fetchContentWithFallback (cidPath, opts = {}) {
+    let lastError = null
+    // we use this to checkpoint at which chunk a request failed.
+    // this is temporary until range requests are supported.
+    let byteCountCheckpoint = 0
+
+    const fetchContent = async function * () {
+      let byteCount = 0
+      const byteChunks = await this.fetchContent(cidPath, opts)
+      for await (const chunk of byteChunks) {
+        // avoid sending duplicate chunks
+        if (byteCount < byteCountCheckpoint) {
+          // checks for overlapping chunks
+          const remainingBytes = byteCountCheckpoint - byteCount
+          if (remainingBytes < chunk.length) {
+            yield chunk.slice(remainingBytes)
```

Comment:
shouldn't byteCountCheckpoint get updated here? (i.e. `byteCountCheckPoint += chunk.length - remainingBytes`)

Reply:
Thanks for catching this Hannah, you are right.

I modified the tests such that it fails if the byteCountCheckPoint isn't updated as you described and now its passing.